disvae/vae.py

```diff
@@ -50,21 +50,28 @@ def reparameterize(self, mean, logvar):
             # Reconstruction mode
             return mean
 
-    def forward(self, x):
+    def forward(self, x, get_sample=False):
         """
         Forward pass of model.
 
         Parameters
         ----------
         x : torch.Tensor
             Batch of data. Shape (batch_size, n_chan, height, width)
+
+        get_sample : bool
+            Returns only the latent sample for factor-VAE
         """
         latent_dist = self.encoder(x)
         latent_sample = self.reparameterize(*latent_dist)
-        reconstruct = self.decoder(latent_sample)
-        if self.is_color:
-            reconstruct = reconstruct * 255
-        return reconstruct, latent_dist
+
+        if get_sample:
+            return latent_sample
```

Comment:
not a big fan of changing the number of outputs of a function using a bool. If it were me i would define a simple helper method in `VAE`:

```
def sample_latent(self, x):
        latent_dist = self.encoder(x)  
        latent_sample = self.reparameterize(*latent_dist)
        return latent_sample
```

I think it's clear from the code what you are doing

Reply:
Agree, I'll do this.